src/recordlinker/linkage/matchers.py

```diff
@@ -0,0 +1,427 @@
+        string1 into string2.
+     - DamerauLevenshtein: the number of edits (including transpositions) needed to
+        transform string1 into string2.
+    :return: The normalized similarity between string1 and string2, with 0 representing
+        no similarity between string1 and string2, and 1 meaning string1 and string2 are
+        dentical words.
+    """
+    if similarity_measure == "JaroWinkler":
+        return rapidfuzz.distance.JaroWinkler.normalized_similarity(string1, string2)
+    elif similarity_measure == "Levenshtein":
+        return rapidfuzz.distance.Levenshtein.normalized_similarity(string1, string2)
+    elif similarity_measure == "DamerauLevenshtein":
+        return rapidfuzz.distance.DamerauLevenshtein.normalized_similarity(
+            string1, string2
+        )
+
+
+def eval_perfect_match(feature_comparisons: list, **kwargs) -> bool:
+    """
+    Determines whether a given set of feature comparisons represent a
+    'perfect' match (i.e. whether all features that were compared match
+    in whatever criteria was specified for them).
+
+    :param feature_comparisons: A list of 1s and 0s, one for each feature
+      that was compared during the match algorithm.
+    :return: The evaluation of whether the given features all match.
+    """
+    return sum(feature_comparisons) == len(feature_comparisons)
+
+
+def eval_log_odds_cutoff(feature_comparisons: list, **kwargs) -> bool:
+    """
+    Determines whether a given set of feature comparisons matches enough
+    to be the result of a true patient link instead of just random chance.
+    This is represented using previously computed log-odds ratios.
+
+    :param feature_comparisons: A list of floats representing the log-odds
+      score of each field computed on.
+    :return: Whether the feature comparisons score well enough to be
+      considered a match.
+    """
+    if "true_match_threshold" not in kwargs:
+        raise KeyError("Cutoff threshold for true matches must be passed.")
+    return sum(feature_comparisons) >= kwargs["true_match_threshold"]
+
+
+def feature_match_exact(
+    record_i: list,
+    record_j: list,
+    feature_col: str,
+    col_to_idx: dict[str, int],
+    **kwargs: dict,
+) -> bool:
+    """
+    Determines whether a single feature in a given pair of records
+    constitutes an exact match (perfect equality).
+
+    :param record_i: One of the records in the candidate pair to evaluate.
+    :param record_j: The second record in the candidate pair.
+    :param feature_col: The name of the column being evaluated (e.g. "city").
+    :param col_to_idx: A dictionary mapping column names to the numeric index
+      in which they occur in order in the data.
+    :return: A boolean indicating whether the features are an exact match.
+    """
+    idx = col_to_idx[feature_col]
+    return record_i[idx] == record_j[idx]
+
+
+def feature_match_four_char(
```

Comment:
What is the purpose of this function when we have transformation functions (e.g., `first4`)?

Reply:
Very similar, but the transformer **alters** data, this function **compares** data.  In issue #10, I'm going to start the process of removing the first4 and last4 transformer functions.  But for now, I'm really just focusing on copying this over from link.py so we can to test the 2 schemas and make sure we're OK to swap our the old one for the new one.  Once we are down to one schema, we'll do some heavy refactoring on this module.